notification_manager.cpp

```diff
@@ -1161,15 +1039,336 @@ string NotificationManager::getPluginInfo(PLUGIN_INFORMATION* info)
 
 /**
  * Create a notification instance
- *                                       
- * NOTE: not yet implemented
- *      
+ *
  * @param    name		The notification to create
  * @param    category		The JSON string with configuration
- * @return			True on success, false otherwise.
+ * @return                      True on success, false otherwise.
  */
 bool NotificationManager::createInstance(const string& name,
 					 const string& category)
 {
+	ConfigCategory config(name, category);
+
+	return this->setupInstance(name, config);
+}
+
+/**
+ * Create and add a new Notification instance to instances map.
+ * Register also interest for configuration changes.
+ *
+ * @param    name		The instance name to create.
+ * @param    config		The configuration for the new instance.
+ * @return			True on success, false otherwise.
+ */
+bool NotificationManager::setupInstance(const string& name,
```

Comment:
This is a very long function, might want to consider splitting it. Having the stuff that extracts/checks configuration separate might be good for reconfiguration.